packages/@aws-cdk/cdk/lib/cloudformation/condition.ts

```diff
@@ -33,3 +33,28 @@ export class Condition extends Referenceable {
     };
   }
 }
+
+/**
+ * You can use intrinsic functions, such as ``Fn::If``, ``Fn::Equals``, and ``Fn::Not``, to conditionally
+ * create stack resources. These conditions are evaluated based on input parameters that you
+ * declare when you create or update a stack. After you define all your conditions, you can
+ * associate them with resources or resource properties in the Resources and Outputs sections
+ * of a template.
+ *
+ * You define all conditions in the Conditions section of a template except for ``Fn::If`` conditions.
+ * You can use the ``Fn::If`` condition in the metadata attribute, update policy attribute, and property
+ * values in the Resources section and Outputs sections of a template.
+ *
+ * You might use conditions when you want to reuse a template that can create resources in different
+ * contexts, such as a test environment versus a production environment. In your template, you can
+ * add an EnvironmentType input parameter, which accepts either prod or test as inputs. For the
+ * production environment, you might include Amazon EC2 instances with certain capabilities;
+ * however, for the test environment, you want to use less capabilities to save costs. With
+ * conditions, you can define which resources are created and how they're configured for each
+ * environment type.
+ */
+export class FnCondition extends CloudFormationToken {
```

Comment:
This should probably carry composition utilities (`.and`, `.or`, ...). I have mixed feelings about `.not` (usually think of it more as a static than instance).

Reply:
I wouldn't want to over index on this programming model. I would rather keep it 1:1 with how CloudFormation expresses it. It is there to allow people to implement 1:1 CloudFormation outputs in the case where they can't really use native programming structures.